app/api/ragie/source/route.ts

```diff
@@ -0,0 +1,49 @@
+import { NextRequest } from "next/server";
+import { z } from "zod";
+
+import { getRagieApiKey } from "@/lib/server/ragie";
+import { RAGIE_API_BASE_URL } from "@/lib/server/settings";
+import { requireAuthContext } from "@/lib/server/utils";
+
+const paramsSchema = z.object({
+  tenant: z.string(),
+  url: z.string(),
+});
+
+export async function GET(request: NextRequest) {
+  const params = paramsSchema.parse({
```

Comment:
I think this will throw an error if the data is invalid and it will probably be a 500, though I could be wrong.

Let's have it do a safeParse, and if there is an error, we return a 422 response.